### DISCUSSION TITLE
The variable name stuff has been merged (ish) and develop is broken

### DISCUSSION THREAD

**Discussion by guillaumevernieres** [NOAA-EMC/GDASApp#280] [Announcements]
### Background
Our develop branch is broken since the merge the of the "variable name" PR's in a few of the JCSDA repositories.

### Temporary solution
I prepared a branch with a JEDI bundle that points to specific commits here:
[https://github.com/NOAA-EMC/GDASApp/tree/feature/pre-ioda](https://github.com/NOAA-EMC/GDASApp/tree/feature/pre-ioda)

**Comment by RussTreadon-NOAA** [comment]
Need to update hash for vader.    `CMakeLists.txt` has

```
 # Core JEDI repositories
   ecbuild_bundle( PROJECT oops  GIT "https://github.com/jcsda/oops.git"  TAG 2ddcd606908d158f282e852483dfc861e84094ec )
   ecbuild_bundle( PROJECT vader GIT "https://github.com/jcsda/vader.git" TAG 64e94e91856954277d8ace4a0b3c455705565d61 )
   ecbuild_bundle( PROJECT saber GIT "https://github.com/jcsda/saber.git" TAG 64e94e91856954277d8ace4a0b3c455705565d61 )
```
Note that vader and saber have same hash.   `build.sh` fails because it can not find the specified vader hash.  Change vader hash to bbc9e9f329b1d93d09e463b2dccbdbfc964d05d4.   Haven't completed build yet so can't guarantee it works.

**Reply by guillaumevernieres** [reply]
Oops, let me go look for that hash @RussTreadon-NOAA . Sorry about that!

**Reply by guillaumevernieres** [reply]
here's the vader commit that I built with @RussTreadon-NOAA : `6a07daff8b893697ebad7e8329427eb9f4a26d19`

**Comment by RussTreadon-NOAA** [comment]
Refreshed my local copy of `feature/pre-ioda`.   Now `build.sh` fails with
```
CMake Error at test/CMakeLists.txt:8 (file):
  file DOWNLOAD HASH mismatch

    for file: [/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/test/sorc/gdas.cd/build/test/gdasapp-fix-782aca.tgz]
      expected hash: [782aca0e0ff579557877129f242d0aa4c4252620233622ecacc1deaa524ca8ed]
        actual hash: [e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
             status: [28;"Timeout was reached"]



CMake Error at test/soca/CMakeLists.txt:8 (file):
  file COPY cannot find
  "/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/test/sorc/gdas.cd/build/test/testdata/iced.2019-04-15-43200.nc":
  No such file or directory.


CMake Error at test/atm/CMakeLists.txt:7 (file):
  file failed to create symbolic link
  '/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/test/sorc/gdas.cd/build/test/testdata/gdas.t06z.adpsfc.tm00.bufr_d':
  No such file or directory

```

I don't understand this message.   A check of `ftp.emc.ncep.noaa.gov` shows that the tarball we want, `gdasapp-fix-782aca.tgz`, is present.   Why does the build say it found e3b0c44?    A manual "ftp ftp.emc.ncep.noaa.gov" hangs.   We seem to be having network issues at NCWCP.   Perhaps this is the real culprit.     I'm stuck.   I can't build `feature/pre-ioda` on Hera.

**Reply by guillaumevernieres** [reply]
Damn. I'm going to try to rebuild that branch from scratch @RussTreadon-NOAA .

**Comment by RussTreadon-NOAA** [comment]
NCWCP network issues are clearing up.   The Orion and Hera builds of `feature/pre-ioda` are currently running and in the component compilation step.

**Comment by RussTreadon-NOAA** [comment]
Hera build completed with `WORKFLOW_BUILD=ON`.    All 40 `test_gdasapp` ctests pass.   Still waiting for Orion build  with `WORKFLOW_BUILD=ON` to finish.

**Comment by ShastriPaturi** [comment]
I could successfully build on hera with feature/pre-ioda for GDASApp and run a fcst.

**Comment by RussTreadon-NOAA** [comment]
The Orion `WORKFLOW_BUILD=ON` build from `feature/pre-ioda` is complete.  All `test_gdasapp` ctests successfully ran to completion.

**Reply by guillaumevernieres** [reply]
Thanks for testing @RussTreadon-NOAA ! Do you want to just copy the hash commits to your branch and we merge the bugfix with your open pr?

**Reply by RussTreadon-NOAA** [reply]
This works @guillaumevernieres , but PR #278 likely won't ready to merge until Monday (1/23).   If we want the updated `CMakeLists.txt` in `develop` today (1/20), a PR based off this issue is the quickest route.   I opt for this.  This is easy and clean.  I can review and approve.
